lib/narou/api.rb

```diff
@@ -6,59 +6,83 @@
 require "open-uri"
 require "zlib"
 require "yaml"
+require "json"
+require "memoist"
 require_relative "../novelinfo"
 
 module Narou
   #
   # 小説家になろうデベロッパーAPI操作クラス
   #
   class API
+    extend Memoist
+
+    # 一度に問い合わせする件数
+    BATCH_LIMIT = 300
+
+    # なろうAPIの結果の圧縮レベル(1〜5)
+    GZIP_LEVEL = 5
+
     #
     # なろうデベロッパーAPIから情報を取得
     #
-    # setting: なろうの SiteSetting
+    # api_url: なろうAPIのエンドポイントを指定。通常小説用と18禁小説用と分かれているため
+    # ncodes: 取得したい小説のNコードを指定。文字列か配列を指定可能
     # of: 出力パラメータ（http://dev.syosetu.com/man/api/#of_parm 参照）
     #
-    def initialize(setting, of)
-      @setting = setting
-      @api_url = @setting["narou_api_url"]
-      @ncode = @setting["ncode"]
-      request_api(of)
+    def initialize(api_url:, ncodes:, of:)
+      @api_url = api_url
+      @ncodes = Array(ncodes).map(&:downcase)
+      @of = "n-#{of}"
+      @splited_of = @of.split("-")
     end
 
     def [](key)
       @api_result[key]
     end
 
-    def request_api(of, gzip = 5)
-      gzip_opt = RUBY_VERSION >= "2.0.0" ? "gzip=#{gzip}&" : ""
-      url = "#{@api_url}?#{gzip_opt}ncode=#{@ncode}&of=#{of}&out=json"
+    def request
```

Comment:
__[method_definition_validator]__
* Lint/DefinitionValidator: request_api is undefined. Did you mean? request at [lib/narou/api.rb#L47](/whiteleaf7/narou/blob/650ef04810edbdc3e4e7d67a350e5ae2ee64ec82/lib/narou/api.rb#L47)

<!-- Comment by [SideCI](https://sideci.com) -->